csrc/tensor_view.cpp

```diff
@@ -1511,10 +1511,11 @@ TensorView* TensorViewBuilder::build() const {
     if (shape_.empty()) {
       *shape_extent = IrBuilder::create<Val>(DataType::Index);
     } else {
-      *shape_extent = maybeCastOp(DataType::Index, shape_.at(i));
+      *shape_extent =
+          SimplifyingIrBuilder::maybeCastExpr(DataType::Index, shape_.at(i));
     }
     IterDomainBuilder builder(FusionGuard::getCurFusion()->zeroVal(), extent);
-    if (extent->isOneInt()) {
+    if (extent->isConstScalar() && extent->evaluateInt() == 1) {
```

Comment:
nit: Should we change `evaluateInt` to do `isConstScalar` and return `std::nullopt` if not?

Reply:
We should just do a single:
```
PolymorphicValue Val::evaluate();
```
and return `std::monostate` if not constant scalar. This is a good first issue tracked at: https://github.com/NVIDIA/Fuser/issues/643